Simplify translations form

JevgenijVisockij opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplify translations form
| Type?         | refacto
| Category?     | BO
| BC breaks?    | yes 
| Deprecations? | no
| Fixed ticket? | Part of #16482
| How to test?  |  Simplify Configure -> International -> Translations. Everything must look the same with 2 exceptions, blue boxes were replaced by help text under and majority of params now are required. 
<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/20243)
<!-- Reviewable:end -->

JevgenijVisockij committed:
Fixed some translation strings

JevgenijVisockij committed:
Fixed domain for translation

JevgenijVisockij committed:
Made changes to copy language has the same design it had before

JevgenijVisockij committed:
CS fixer fixes

matks reviewed (dismissed):
Looks good 👍  is it still WIP ?

Julievrz commented:
Wording looks good to me, thanks! 👍

Julievrz commented:
> This pull request seems to contain new translation strings. I have summarized them below to ease up review:
> 
> * `Admin.International.Notification`
>   
>   * [ ]  [`Theme from which translations will be copied.`](https://github.com/PrestaShop/PrestaShop/pull/21527/files#diff-fb6de229b5e6f5962f30158b3e94da23R82)
>   * [ ]  [`Theme to which translations will be copied.`](https://github.com/PrestaShop/PrestaShop/pull/21527/files#diff-fb6de229b5e6f5962f30158b3e94da23R99)
> 
> (Note: this is an automated message, but answering it will reach a real human)

I couldn't find this wording in the PR but I assumed it was because you canceled it, am I right?

JevgenijVisockij commented:
>     * Theme from which translations will be copied.

Yeah that must be the case.

JevgenijVisockij commented:
> > This pull request seems to contain new translation strings. I have summarized them below to ease up review:
> > 
> > * `Admin.International.Notification`
> >   
> >   * [ ]   [`Theme from which translations will be copied.`](https://github.com/PrestaShop/PrestaShop/pull/21527/files#diff-fb6de229b5e6f5962f30158b3e94da23R82)
> >   * [ ]   [`Theme to which translations will be copied.`](https://github.com/PrestaShop/PrestaShop/pull/21527/files#diff-fb6de229b5e6f5962f30158b3e94da23R99)
> > 
> > (Note: this is an automated message, but answering it will reach a real human)
> 
> I couldn't find this wording in the PR but I assumed it was because you canceled it, am I right?

Yep, that must be the case.

JevgenijVisockij commented:
> Looks good +1 is it still WIP ?

Yeah, I still want to change this form so it always renders from instead of redirecting and remove use of @Template

JevgenijVisockij committed:
All forms in translation form are checked for validity and that from renders form upon submission instead of redirecting

JevgenijVisockij commented:
This is no longer WIP

Progi1984 commented:
> This is no longer WIP

Label removed. Thanks for the warning.

Progi1984 commented on src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/copy_language.html.twig:
```diff
@@ -80,8 +82,7 @@
     </div>
-
-    <div class="card-footer">
+        <div class="card-footer">
```
```suggestion
    <div class="card-footer">
```

Progi1984 reviewed (commented):
A small change and it's approved for me

JevgenijVisockij committed:
Update src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/copy_language.html.twig

Fixed divs position

Co-authored-by: Progi1984 <progi1984@gmail.com>

khouloudbelguith commented:
Hi @JevgenijVisockij,

Please check the Conflicting files.

Thank you!

JevgenijVisockij committed:
Simplify translations form

# Conflicts:
#	src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/add_update_language.html.twig
#	src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/export_language.html.twig
#	src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/modify_translations.html.twig

JevgenijVisockij commented:
> Hi @JevgenijVisockij,
> 
> Please check the Conflicting files.
> 
> Thank you!

Done.

SD1982 commented:
Hi @JevgenijVisockij  I detected a few things during the PR test (no finished yet)
**In the "header" before the PR**
![image](https://user-images.githubusercontent.com/40635814/104927326-96d3c600-59a1-11eb-90af-8d5d2d261600.png)
**In the "header" after the PR**
![image](https://user-images.githubusercontent.com/40635814/104927436-b4089480-59a1-11eb-819f-8905adb3eb82.png)
**In the "modifiy translation form"**
![image](https://user-images.githubusercontent.com/40635814/104927599-ee723180-59a1-11eb-93c7-e98e6e114c3d.png)

JevgenijVisockij committed:
Reverted the controller changes

JevgenijVisockij commented:
> Hi @JevgenijVisockij I detected a few things during the PR test (no finished yet)
> **In the "header" before the PR**
> ![image](https://user-images.githubusercontent.com/40635814/104927326-96d3c600-59a1-11eb-90af-8d5d2d261600.png)
> **In the "header" after the PR**
> ![image](https://user-images.githubusercontent.com/40635814/104927436-b4089480-59a1-11eb-819f-8905adb3eb82.png)
> **In the "modifiy translation form"**
> ![image](https://user-images.githubusercontent.com/40635814/104927599-ee723180-59a1-11eb-93c7-e98e6e114c3d.png)

Hi. Thanks for Q&A. 

Regarding first problem I couldn't confirm that this is caused by PR. I too had "Main Country No orders" message at first, but it didn't seem to matter if I swtiched between branches. 

After changing state of one of the orders to "delivered" it now shows "100 United states". Is it possible that during no PR attempt you had orders which had invoices available(depends on order state, you can check Order Settings -> Statuses). And during the attempt with PR you didn't change any statuses on orders, thus "No orders" result?

Regarding second problem I could not reproduce it. Could you provide more information? (browser, screen size).

matks commented:
@SD1982 (or another QA analyst)

Just like @JevgenijVisockij I do not succeed to reproduce the QA issues mentioned.

For the 1st, I checked: the number of orders come from another page (URL is `/index.php?controller=AdminStats&ajax=1&action=getKpi&kpi=main_country&token=XXX&rand=1611767466028&_=YYY`)

For the 2nd, it might have been CSS badly cached by browser?
![Capture d’écran 2021-01-27 à 17 59 53](https://user-images.githubusercontent.com/3830050/106027786-858a6800-60cb-11eb-8260-fee4f3eac55b.png)

Robin-Fischer-PS commented:
Hi @JevgenijVisockij !

I have 2 questions on your PR : 

- The label alignment has changed with your PR (left alignment without the PR, right alignment with the PR). Is this a wanted behavior ?
_Without the PR :_ 
![pr 21527 left alignment before](https://user-images.githubusercontent.com/55688534/106267035-6570bb00-6229-11eb-8f49-f248c1512ea0.png)
_With the PR :_ 
![pr 21527 right alignment after](https://user-images.githubusercontent.com/55688534/106267109-80432f80-6229-11eb-8d66-191aa36161c9.png)

- The helpbox in "Copy" block is still displayed : Is this wanted too ? (I think so since there is pretty important information in it)
![pr 21527 copy helpbox](https://user-images.githubusercontent.com/55688534/106267288-b7b1dc00-6229-11eb-94f3-96999f7d9e59.png)

Thanks !

JevgenijVisockij commented:
> Hi @JevgenijVisockij !
> 
> I have 2 questions on your PR :
> 
>     * The label alignment has changed with your PR (left alignment without the PR, right alignment with the PR). Is this a wanted behavior ?
>       _Without the PR :_
>       ![pr 21527 left alignment before](https://user-images.githubusercontent.com/55688534/106267035-6570bb00-6229-11eb-8f49-f248c1512ea0.png)
>       _With the PR :_
>       ![pr 21527 right alignment after](https://user-images.githubusercontent.com/55688534/106267109-80432f80-6229-11eb-8d66-191aa36161c9.png)
> 
>     * The helpbox in "Copy" block is still displayed : Is this wanted too ? (I think so since there is pretty important information in it)
>       ![pr 21527 copy helpbox](https://user-images.githubusercontent.com/55688534/106267288-b7b1dc00-6229-11eb-94f3-96999f7d9e59.png)
> 
> 
> Thanks !

Hey, thanks for Q&A!

Regarding first question, I didn't specifically aim to do it, it was a consequence of form simplification, but I don't see any problems with it, after all labels is is right aligned in all other forms.
@PrestaShop/product-team WTDY? Is this okay or is it important that labels stay left-aligned in this case?

Regarding second question. I am guessing you are asking because of "lue boxes were replaced by help text under" in the description? It's meant to only replace the blue help boxes (well it's no longer a box after design update, so not sure how to call it) where you need to hover over
![image](https://user-images.githubusercontent.com/47781757/106745355-a56cdf00-6629-11eb-9c86-50af55fdb98e.png)
The larger help boxes like you mentioned shouldn't be affected.

Progi1984 commented:
Ping @NeOMakinG

prestascott commented:
Hello everyone and thank you for your work!

@JevgenijVisockij It's a UX question right here so I'll answer in the name of the Product Team. To keep consistency with the back office pages, we must right-align label text and leave the red asterisk on the left side of the label. 

Otherwise, everything is good to me. Thanks for this great PR :)

Robin-Fischer-PS commented:
Since it's OK for UX, everything is fine for me, it's QA :heavy_check_mark: !

Thanks @JevgenijVisockij :)

Progi1984 commented:
Thanks @JevgenijVisockij & @Robin-Fischer-PS
